tests/performance/test_performance.py

```diff
@@ -83,8 +99,55 @@ def _validate_metric_against_threshold(actual, expected_unit, size):
     assert actual < size * expected_unit
 
 
-@pytest.mark.parametrize('dataset_generator', dataset_generators)
-def test_performance(dataset_generator):
+def validate_performance(transformer, dataset_generator, should_assert=False, results=None):
+    """Validate the performance of all transformers for a dataset_generator.
+
+    Args:
+        transformer (rdt.transformers.BaseTransformer):
+            The transformer to evaluate.
+        dataset_generator (rdt.tests.datasets.BaseDatasetGenerator):
+            The dataset generator to performance tests against.
+        should_assert (bool):
+            Whether or not to raise AssertionErrors.
+        results (dict or None):
+            If not None, performance results will be collected and stored in results.
+    """
+    transformer_instance = transformer()
+    expected = dataset_generator.get_performance_thresholds()
+
+    dataset_sizes = _get_dataset_sizes(dataset_generator.DATA_TYPE)
+    for sizes in dataset_sizes:
```

Comment:
I think that this should be simplified a bit further by re-organizing some things:
- One function should just run the loop over the sizes and produce the maximum metric relative to the size.
- Another function validates the obtained value against the specific thresholds. This can happen only once by metric using the maximum relative value, since we are only interested on whether the maximum is valid or not. This function is the one that is told whether to assert or not.
- Another function, which is the actual test, calls both of the previous functions, telling the second one to assert.
- Another function, which is the user-facing function, calls both of the functions above without asserting, and then aggregates, etc.

Inside `test_performance.py`

```python3
def evaluate_transformer_performance(transformer, dataset_generator):
    transformer_instance = transformer()

    sizes = _get_dataset_sizes(dataset_generator.DATA_TYPE)

    out = []
    for fit_size, transform_size in sizes:
        performance = profile_transformer(
            transformer=transformer_instance,
            dataset_generator=dataset_generator,
            fit_size=fit_size,
            transform_size=transform_size,
        )
        size = np.array([fit_size, transform_size, transform_size] * 2)
        out.append(performance / size)

    return pd.DataFrame(out).max(axis=0)


def validate_performance(performance, dataset_generator, should_assert):
    expected = dataset_generator.get_performance_thresholds()
    out = []
    for test_name, value in performance.items():
        function, metric = test_name.lower().replace(' ', '_').rsplit('_', 1)
        valid = value < expected[function][metric]
        out.append(valid)
        if should_assert and not valid:
            raise AssertionError()

    return out


@pytest.mark.parametrize(('transformer', 'dataset_generator'), test_cases)
def test_performance(transformer, dataset_generator):
    performance = evaluate_transformer_performance(transformer, dataset_generator)
    validate_performance(performance, dataset_generator, should_assert=True)
```

Inside `contributing.py` (See comments above too):

```python
def validate_transformer_performance(desired_transformer):
    ...
    total_results = pd.DataFrame()
    for transformer in transformers:
        for dataset_generator in dataset_generators:
            performance = evaluate_transformer_performance(transformer, dataset_generator)
            valid = validate_performance(performance, dataset_generator, should_assert=False)

            results = pd.DataFrame({
                'Value': performance.to_numpy(),
                'Valid': valid,
                'transformer': transformer.__name__,
                'dataset': dataset_generator.__name__,
            })
            results['Evaluation Metric'] = performance.index
            total_results = total_results.append(results)

    ... aggregate...
```